grails-app/controllers/com/unifina/controller/api/UserApiController.groovy

```diff
@@ -1,29 +1,100 @@
 package com.unifina.controller.api
 
+import com.unifina.api.ApiException
 import com.unifina.domain.security.SecUser
 import com.unifina.security.StreamrApi
 import com.unifina.service.UserService
 import grails.converters.JSON
 import grails.plugin.springsecurity.annotation.Secured
+import grails.validation.Validateable
 
 @Secured(["IS_AUTHENTICATED_ANONYMOUSLY"])
 class UserApiController {
 	static allowedMethods = [
+		update: "PUT",
+		changePassword: "POST",
 		getUserInfo: "GET",
 		delete: "DELETE",
 	]
 
+	def springSecurityService
 	UserService userService
 
+	@StreamrApi
+	def update(UpdateProfileCommand cmd) {
+		SecUser user = loggedInUser()
+		// Only these user fields can be updated!
+		user.name = cmd.name ?: user.name
+		user = user.save(failOnError: true)
+		if (user.hasErrors()) {
+			log.warn("Update failed due to validation errors: " + userService.checkErrors(user.errors.getAllErrors()))
+			throw new ApiException(400, "PROFILE_UPDATE_FAILED", "Profile update failed.")
+		}
+		return render(user.toMap() as JSON)
+	}
+
+	@StreamrApi
+	def changePassword(ChangePasswordCommand cmd) {
+		if (!cmd.validate()) {
+			throw new ApiException(400, "PASSWORD_CHANGE_FAILED", "Password not changed!")
+		}
+		SecUser user = loggedInUser()
+		user.password = springSecurityService.encodePassword(cmd.password)
+		user.save(flush: true, failOnError: true)
+		springSecurityService.reauthenticate(user.username)
```

Comment:
The new frontend doesn't use the `springSecurityService` cookie-based sessions anymore, so I think this line can (and should) be removed (if the old frontend does not use this endpoint).